assets/src/components/Lightbox/Lightbox.js

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import PhotoSwipe from '../../../../node_modules/photoswipe/dist/photoswipe.js';
+import PhotoSwipeUI_Default from '../../../../node_modules/photoswipe/dist/photoswipe-ui-default.js';
+
+export const setupLightboxForImageBlocks = function() {
+  const images = document.querySelectorAll('.wp-block-image img');
+  const items = [...images].map(image => ({
+    src: image.src ? image.src : image.dataset.src,
+    w: 0,
+    h: 0
+  }));
+
+  const imageBlocks = document.querySelectorAll('.wp-block-image');
+  imageBlocks.forEach((imageBlock, index) => {   
+    const caption = imageBlock.querySelector('figcaption');
+    if (caption) {
+      items[index].title = caption.innerHTML;
+    }
+
+    imageBlock.querySelector('img').addEventListener('click', () => wp.element.render(
+        <Lightbox isOpen={true} index={index} items={items} onClose={() => {}} />,
+        document.body.appendChild(document.createElement('div'))
```

Comment:
Seems hacky? it is... I did this because the `createPortal` call is now inside the LightBox component, but I'll extract that into a `<Modal />` component, wrap the LightBox with `Modal` in the Gallery components, and use `createPortal` here instead of `render`.